tests/test_pso.c

```diff
@@ -36,14 +36,12 @@ Test(pso_unit,pso_eval_fitness){
    10.0,  3.0,
   -10.0, -3.0
   };
-  double * fitness = (double*)malloc(sizeof(double)*swarm_size);
+  double fitness[sizeof(double)*swarm_size];

```

Comment:
You need to remove all sizeof stuff in the static assignments. In static assignment, the type of the array is given so you don't need to specify size of the individual elements. You do this is many places.

Reply:
oh fuck. completly missed that xD